crates/integrations/datafusion/tests/integration_datafusion_test.rs

```diff
@@ -810,3 +814,147 @@ async fn test_insert_into_nested() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_insert_into_partitioned() -> Result<()> {
```

Comment:
Could we move this test to newly added sqllogictests? I think we should be able to read/write now.

Reply:
I think the overall direction is to remove integration tests like this, and move them in sqllogictests. But we could delete this after we fix #1835 . 

>This test case still has its value as it validates the inserted data files are put under the partitioned path correctly.

I'm not convinced. Such detailed check could be done in ut of low level apis rather in integration tests.